client/share_trader/js/components/share_trader.js

```diff
@@ -50,26 +47,61 @@ const ShareTrader = React.createClass({
         this.setState(state);
     },
 
-    setActiveAccount(account) {
-        this.setState({
-            activeAccount: account
+    fetchAssetList(account) {
+        AssetActions.flushAssetList();
+        const { activeAccount } = this.state;
+
+        /* fetch all assets */
+        let accountPublicKey = null;
+
+        if (account || activeAccount) {
+            accountPublicKey = account ? account.vk : activeAccount.vk;
+        }
+        accountPublicKey = account === null ? null : accountPublicKey;
```

Comment:
it's somewhat tricky
`account` can be `object` or `undefined` or `null`
if it's `object` or `null`, use `account`
otherwise `this.state.activeAccount` (which may be `null`)

suggestions?